daemon/containerd/image_prune.go

```diff
@@ -107,8 +108,16 @@ func (i *ImageService) pruneUnused(ctx context.Context, filterFunc imageFilterFu
 
 	usedDigests := filterImagesUsedByContainers(ctx, i.containers.List(), imagesToPrune)
 
+	// Sort images by name to make the behavior deterministic and consistent with graphdrivers.
+	sorted := make([]string, 0, len(imagesToPrune))
+	for name := range imagesToPrune {
+		sorted = append(sorted, name)
+	}
```

Comment:
We can probably start looking at the `maps` package, and for these kind of things, use [`maps.Keys()`](https://pkg.go.dev/golang.org/x/exp/maps#Keys)